src/main/java/seedu/address/model/lesson/MakeUpLesson.java

```diff
@@ -28,6 +28,14 @@ public boolean isRecurring() {
         return false;
     }
 
+    /**
+     * This method should not be called since MakeUpLesson only has one date.
+     */
+    @Override
+    public Date getNextDate() {
+        throw new UnsupportedOperationException("Makeup lesson has no upcoming date.");
+    }
```

Comment:
This violates LSP, see: https://www.baeldung.com/java-liskov-substitution-principle

Can we make the method name sound more general such that we can just return start date here?

Reply:
I will change it to getDisplayDate(), what do you think?